pkg/driver/controller.go

```diff
@@ -591,39 +610,15 @@ func (d *Driver) ControllerExpandVolume(ctx context.Context, req *csi.Controller
 		}, nil
 	}
 
+	// Run a no-op because volume expansion is not supported for child volumes
 	if splitVolumeId[0] == cloud.VolumePrefix {
-		v, err := d.cloud.DescribeVolume(ctx, volumeID)
-		if err != nil {
-			if err == cloud.ErrNotFound {
-				return nil, status.Errorf(codes.NotFound, "Volume not found with ID %q", volumeID)
-			}
-			return nil, status.Errorf(codes.Internal, "Could not get volume with ID %q: %v", volumeID, err)
-		}
-
-		if newCapacity <= v.StorageCapacityQuotaGiB {
-			// Current capacity is sufficient to satisfy the request
-			klog.V(4).Infof("ControllerExpandVolume: current volume capacity of %d GiB matches or exceeds requested storage capacity of %d GiB, returning with success", v.StorageCapacityQuotaGiB, newCapacity)
-			return &csi.ControllerExpandVolumeResponse{
-				CapacityBytes:         util.GiBToBytes(v.StorageCapacityQuotaGiB),
-				NodeExpansionRequired: false,
-			}, nil
-		}
-
-		finalCapacity, err := d.cloud.ResizeVolume(ctx, volumeID, newCapacity)
-		if err != nil {
-			return nil, status.Errorf(codes.Internal, "resize failed: %v", err)
-		}
-
-		err = d.cloud.WaitForVolumeResize(ctx, volumeID, *finalCapacity)
-		if err != nil {
-			return nil, status.Errorf(codes.Internal, "volume is not resized: %v", err)
-		}
-
+		klog.V(2).InfoS("Volume expansion is not supported for child volumes. Running no-op and returning success.")
```

Comment:
Not sure how I feel about this. We should probably throw an `Unimplemented` error instead. If a user forgets that they can't scale storage an error will be more noticeable and makes the most sense in my opinion.